Core/clim-basic/extended-streams/text-formatting.lisp

```diff
@@ -40,17 +53,13 @@
 
 (defmethod initialize-instance :after ((instance standard-page-layout)
                                        &key text-margins text-margin)
-  (macrolet ((thunk (edge default)
-               `(let ((spec (getf text-margins ,edge)))
-                  (typecase spec
-                    (null (setf (getf text-margins ,edge) ,default))
-                    (atom (setf (getf text-margins ,edge) `(:relative ,spec)))))))
-    (thunk :left   `(:relative 0))
-    (thunk :top    `(:relative 0))
-    (thunk :right  (if text-margin
-                       `(:absolute ,text-margin)
-                       `(:relative 0)))
-    (thunk :bottom `(:relative 0))
+  (let ((right-margin (if text-margin
+                          `(:absolute ,text-margin)
+                          `(:relative 0))))
+    (setf text-margins (normalize-margin-spec text-margins `(:left   (:relative 0)
+                                                             :top    (:relative 0)
+                                                             :right  ,right-margin
+                                                             :bottom (:relative 0))))
     (setf (slot-value instance 'margins) text-margins)))
```

Comment:
You could just `(setf (slot-value instance 'margins) (normalize …))` now.

Reply:
done